warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -97,6 +99,10 @@ public UniqueTransform(BaseQueryLogic<Entry<Key,Value>> logic, Set<String> field
     @Override
     public Entry<Key,Document> apply(@Nullable Entry<Key,Document> keyDocumentEntry) {
         if (keyDocumentEntry != null) {
+            if (isTimingMetadata(keyDocumentEntry.getValue())) {
```

Comment:
I don't think this is exactly what we need to do.  If we are collecting timing details then every document will have timing metadata (see PipelineQuerySpanCollectionIterator) in which case everything will be returned defeating the whole purpose of this transform.  What you actually need to do here is ignore the "Final Document" which is the entry that contains a summary after all documents are returned.  To test for that use FinalDocumentTrackingIterator.isFinalDocumentKey(Key k).

Reply:
Made changes per comment. Please see latest commit.